Point_set_processing_3/include/CGAL/poisson_eliminate.h

```diff
@@ -0,0 +1,383 @@
+  double r_min = CGAL::to_double(weight_limiting ? internal::get_minimum_radius(points.size(), number_of_points, beta, gamma, r_max) : 0);
+
+  auto weight_functor = parameters::choose_parameter(parameters::get_parameter(np, internal_np::weight_function), internal::Weight_functor<Point>(r_min, alpha));
+
+  std::size_t heap_size = points.size();
+  std::vector<Point> tiling_points;
+
+  IPM ipm(points, tiling_points, point_map);
+
+  auto tile_point = [&tiling_points, &bb, &r_max, &ambient_dimension](const Point& p, std::size_t dim = 0) {
+    auto do_tiling = [&tiling_points, &bb, &r_max, &ambient_dimension](const auto& self, const Point& p, std::size_t dim) -> void {
+      auto it = p.cartesian_begin();
+
+      if (bb.min_coord(int(dim)) > (*(it + dim) - r_max)) {
+        FT v = 2 * bb.min_coord(int(dim)) - (*(it + dim));
+        Point p2;
+        internal::copy_and_replace(p, p2, dim, v);
+        tiling_points.emplace_back(p2);
+        if (dim + 1 < ambient_dimension)
+          self(self, tiling_points.back(), dim + 1);
+      }
+
+      if (bb.max_coord(int(dim)) < (*(it + dim) + r_max)) {
+        FT v = 2 * bb.max_coord(int(dim)) - (*(it + dim));
+        Point p2;
+        internal::copy_and_replace(p, p2, dim, v);
+        tiling_points.emplace_back(p2);
+        if (dim + 1 < ambient_dimension)
+          self(self, tiling_points.back(), dim + 1);
+      }
+
+      if (dim + 1 < ambient_dimension)
+        self(self, p, dim + 1);
+      };
+    do_tiling(do_tiling, p, dim);
+    };
+
+  if (tiling) {
+    for (std::size_t i = 0;i<points.size();i++)
+      tile_point(get(point_map, points[i]));
+  }
+
+  // Tiling requires either to copy the point range or to use a special Index_property_map with a second array of points to keep the input range const.
+  Tree tree(boost::counting_iterator<std::size_t>(0), boost::counting_iterator<std::size_t>(points.size() + tiling_points.size()), Splitter(), Search_traits(ipm));
+  tree.build();
+
+  std::vector<std::size_t> heap(boost::counting_iterator<std::size_t>(0), boost::counting_iterator<std::size_t>(points.size()));
+  std::vector<std::size_t> heap_pos(points.size());
+
+  std::size_t target_points = number_of_points;
+
+  do {
+    // Computing weights
+    std::vector<double> weights(points.size(), 0);
+    std::vector<std::size_t> res;
+    for (std::size_t i = 0; i < heap_size; i++) {
+      const Point& p = get(ipm, heap[i]);
+      res.clear();
+      tree.search(std::back_inserter(res), Fuzzy_sphere(p, r_max, 0, Search_traits(ipm)));
+
+      for (std::size_t n = 0; n < res.size(); n++) {
+        if (i == res[n] || res[n] >= heap_pos.size() || heap_pos[res[n]] >= heap_size)
+          continue;
+
+        const Point p2 = get(ipm, res[n]);
+        double d2 = CGAL::to_double((p - p2).squared_length());
```

Comment:
Do we have an `approximate_something()` ?? @sloriot

Reply:
CGAL::approximate_sqrt

Reply:
I was more wondering if we have an `approximate_squared_distance(p,q)`

Reply:
I am now using Kernel::compute_squared_distance_(2/3)_object or squared_distance_d_object depending on the Ambient_dimension of the Kernel.